File: simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/HomeKey.java
package com.zhuinden.navigationexamplefrag;

import com.google.auto.value.AutoValue;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class HomeKey extends BaseKey {
    public static HomeKey create() {
        return new AutoValue_HomeKey();
    }

    @Override
    protected BaseFragment createFragment() {
        return new HomeFragment();
    }
}

File: simple-stack-example-kotlin/src/main/kotlin/com/zhuinden/simplestackexamplekotlin/HomeKey.kt
package com.zhuinden.simplestackexamplekotlin

import android.os.Parcelable
import paperparcel.PaperParcel

/**
 * Created by Owner on 2017. 06. 29..
 */
@PaperParcel
object HomeKey : BaseKey() {
    override fun createFragment() = HomeFragment()

    @JvmField val CREATOR: Parcelable.Creator<HomeKey> = PaperParcelHomeKey.CREATOR

    override fun toString(): String = javaClass.name // you NEED to implement this in an `object` because of Kotlin!
}
